superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -116,36 +122,87 @@ export default function DrillByModal({
   filters,
   formData,
   groupbyFieldName = 'groupby',
+  adhocFilterFieldName = 'adhoc_filters',
   onHideModal,
 }: DrillByModalProps) {
   const theme = useTheme();
-  const [chartDataResult, setChartDataResult] = useState<QueryData[]>();
-  const [drillByDisplayMode, setDrillByDisplayMode] = useState<DrillByType>(
-    DrillByType.Chart,
+
+  const initialGroupbyColumns = useMemo(
+    () =>
+      ensureIsArray(formData[groupbyFieldName])
+        .map(colName =>
+          dataset.columns?.find(col => col.column_name === colName),
+        )
+        .filter(isDefined),
+    [dataset.columns, formData, groupbyFieldName],
   );
+
+  const { displayModeToggle, drillByDisplayMode } = useDisplayModeToggle();
+  const [chartDataResult, setChartDataResult] = useState<QueryData[]>();
   const [datasourceId] = useMemo(
     () => formData.datasource.split('__'),
     [formData.datasource],
   );
 
-  const [currentColumn, setCurrentColumn] = useState(column);
+  // currentColumn can be an array when the original chart is grouped by multiple
+  // columns and user navigates back to the original chart by clicking the first
+  // breadcrumb
+  const [currentColumn, setCurrentColumn] = useState<
+    Column | Column[] | undefined
+  >(column);
```

Comment:
Just a thought - would it potentially simplify the code, if we'd do something like this (we'd also do something similar in those other similar code blocks):
```typescript
  const [currentColumns, setCurrentColumns] = useState(ensureIsArray(column));
```
The we wouldn't have to jump between `currentColumn` and `currentColumns` or worry about it potentially being `undefined`.